core/kogitobuild/status.go

```diff
@@ -64,114 +44,136 @@ func NewStatusHandler(context *operator.Context) StatusHandler {
 }
 
 func (s *statusHandler) HandleStatusChange(instance api.KogitoBuildInterface, err error) {
-	needUpdate := false
-	sortConditionsByTransitionTime(instance)
+	instance.GetStatus().SetConditions(&[]metav1.Condition{})
 	if err != nil {
-		needUpdate = true
-		addConditionError(instance, err)
+		s.setFailedConditions(instance.GetStatus().GetConditions(), api.OperatorFailureReason, err.Error())
 	} else {
-		if needUpdate, err = handleConditionTransition(instance, s.Client); err != nil {
+		if err = s.handleConditionTransition(instance); err != nil {
 			s.Log.Error(err, "Failed to update build status")
 		}
 	}
-	trimConditions(instance)
-	if needUpdate {
-		if err = s.updateStatus(instance); err != nil {
-			s.Log.Error(err, "Failed to update KogitoBuild")
-		}
+	if err = s.updateStatus(instance); err != nil {
+		s.Log.Error(err, "Failed to update KogitoBuild")
 	}
 }
 
-func sortConditionsByTransitionTime(instance api.KogitoBuildInterface) {
-	sort.SliceStable(instance.GetStatus().GetConditions(), func(i, j int) bool {
-		firstTransitionTime := instance.GetStatus().GetConditions()[i].GetLastTransitionTime()
-		secondTransitionTime := instance.GetStatus().GetConditions()[j].GetLastTransitionTime()
-		return firstTransitionTime.Before(&secondTransitionTime)
-	})
+// newSuccessfulCondition ...
+func (s *statusHandler) newSuccessfulCondition(status metav1.ConditionStatus) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(api.KogitoBuildSuccessful),
+		Status:             status,
+		LastTransitionTime: metav1.Now(),
+		Reason:             string(status),
```

Comment:
Results in:
```
    - lastTransitionTime: '2021-03-18T08:46:42Z'
      message: ''
      reason: 'False'
      status: 'False'
      type: Successful
```
IMHO we should provide some meaningful reason like BuildInProgress (for the case when build is still in progress), possibly some error reason too for the case of a build error.